modules/chmgr/chmgr.js

```diff
@@ -126,379 +126,401 @@ async function check_role(user, guild, cmd) {
+		await bot.api.emb(
+			"Nicht gefunden",
+			"Konnte keine Category-Channel in der Datenbank finden, die dir gehören.",
+			message.channel
+		);
+		return;
+	}
+}
 
-			const remove = res.params["-remove"][0];
-			let guildMember;
-			try {
-				const user_str = res.params["-name"][0];
-				guildMember = await message.guild.members.fetch(
-					user_str.substring(3, user_str.length - 1)
-				);
-			} catch (error) {
-				await bot.api.emb("Falscher User", "Der User ist mir nicht bekannt.");
-				return;
-			}
+async function handleInvite(message, res) {
+	await check_role(message.author, message.guild, "invite");
+	let index, category;
+	const thischannel = await message.channel.fetch(true); //get the full version of this channel
+	/*check, if user is owner of this channel and get the parent category of it, if its a child*/
+	try {
+		index = bot.api.lookup_key_value(
+			databases[0].name,
+			"channelID",
+			message.channel.id
+		)[0];
+		const owner = bot.api.lookup_index(
+			databases[0].name,
+			index,
+			"ownerID"
+		);
+		if (owner != message.author.id) throw new Error();
+		if (thischannel.type != "category") {
+			//could be replaced by channel.type == 'category' because there are only top level categories.
+			category = thischannel.parent;
+		} else {
+			category = thischannel;
+		}
+	} catch (error) {
+		await bot.api.emb(
+			"Berechtigungsfehler",
+			"Du bist nicht der Owner des Channels."
+		);
+		return;
+	}
 
-			if (
-				bot.api.lookup_index(databases[0].name, index, "manage_type") ==
-					"role"
-			) {
-				category.permissionOverwrites.each(async (r) => {
-					if (r.type == "role" && r.id != message.guild.roles.everyone.id) {
-						if (
-							remove == "false" &&
-								guildMember.roles.cache.has(r.id) == false
-						) {
-							guildMember.roles.add(r.id);
-							await bot.api.emb(
-								"Erfolgreich",
-								`${guildMember.toString()} ist jetzt per Rolle eingeladen.`,
-								message.channel
-							);
-							return;
-						} else if (
-							remove == "true" &&
-								guildMember.roles.cache.has(r.id) == true
-						) {
-							guildMember.roles.remove(r.id);
-							await bot.api.emb(
-								"Erfolgreich",
-								`${guildMember.toString()} ist jetzt per Rolle ausgeladen.`,
-								message.channel
-							);
-							return;
-						} else {
-							await bot.api.emb(
-								"Fehler",
-								`${guildMember.toString()} wurde entweder schon entfernt oder hinzugefügt.`,
-								message.channel
-							);
-						}
-					}
-				});
-			} else {
-				if (remove == "false") {
-					const permissions = bot.api.config_get(
-						attributes,
-						message.guild.id,
-						"area_role_attributes"
-					);
-					category.updateOverwrite(guildMember, permissions);
+	const remove = res.params["-remove"][0];
+	let guildMember;
+	try {
+		// user_str can start with <@ or <@!, so we should use a regex here
+		const user_str = res.params["-name"][0].replace( /^\D+/g, "").slice(0, -1);
```

Comment:
when mentioning users, user_str can either be <@!...> when the user is mentioned using their nickname or <@...> when the user is mentioned with their discord name.